runtime/asset_font_selector.cc

```diff
@@ -116,14 +116,19 @@ AssetFontSelector::FlutterFontAttributes::FlutterFontAttributes(
 
 AssetFontSelector::FlutterFontAttributes::~FlutterFontAttributes() {}
 
+AssetProvider* AssetFontSelector::asset_provider() {
+  if (directory_asset_bundle_) {
+    return directory_asset_bundle_.get();
+  } else {
+    return asset_store_.get();
+  }
+}
+
 void AssetFontSelector::parseFontManifest() {
   std::vector<uint8_t> font_manifest_data;
-  if (!directory_asset_bundle_ ||
-      !directory_asset_bundle_->GetAsBuffer(kFontManifestAssetPath,
-                                            &font_manifest_data)) {
-    if (!asset_store_ ||
-        !asset_store_->GetAsBuffer(kFontManifestAssetPath, &font_manifest_data))
-      return;
+  if (!asset_provider()->GetAsBuffer(kFontManifestAssetPath,
```

Comment:
I don't think this behaves exactly like before. The previous code would first look in the directory, and then in the zip file if it wasn't there (or the directory wasn't configured). The new code only looks in either the directory or the zip file, with no fallback.

Reply:
An AssetFontSelector can be constructed from either a DirectoryAssetBundle or a ZipAssetStore.  Only one of those fields will be populated.